src/vmm/src/io_uring/queue/completion.rs

```diff
@@ -93,7 +92,13 @@ impl CompletionQueue {
             self.unmasked_head += Wrapping(1u32);
             ring.store(self.unmasked_head.0, self.head_off, Ordering::Release)?;
 
-            Ok(Some(Cqe::new(cqe)))
+            let res = cqe.res;
+            #[allow(clippy::cast_possible_truncation)]
+            let index = cqe.user_data as usize;
+            #[allow(clippy::unnecessary_literal_unwrap)]
+            let user_data = slab.try_remove(index).unwrap_or_else(|| None::<T>.unwrap());
```

Comment:
Here as well the return of `try_remove` should either be converted into a error or unwrapped with `expect` if error can not be handled.